checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -1,176 +1,873 @@
+   * @param invocation the method invocation node to check
+   * @param varArgsNode the VarArg node of the given method invocation node
+   * @param arg the argument of a method invocation node
+   */
+  private void analyzeCalledMethodsSetForVarArgsToComputeOwningParams(
+      Set<Obligation> obligations,
+      List<? extends VariableTree> paramsOfCurrentMethod,
+      MethodInvocationNode invocation,
+      ArrayCreationNode varArgsNode,
+      Node arg) {
+    for (Node varArgNode : varArgsNode.getInitializers()) {
+      Element varArgElt = TreeUtils.elementFromTree(varArgNode.getTree());
+
+      if (varArgElt == null) {
+        continue;
+      }
+
+      if (varArgElt.getKind().isField()) {
+        inferOwningField(varArgNode, invocation);
+      } else {
+        analyzeCalledMethodsSetForArgAliasesToInferOwningParams(
+            obligations, paramsOfCurrentMethod, invocation, arg);
+      }
+    }
+  }
+
+  /**
+   * For any parameter of the current method that is aliased with an argument passed to the method
+   * invocation, check the set of called methods for the parameter after the call, in order to infer
+   * the owning annotation for that parameter.
+   *
+   * @param obligations set of obligations associated with the current block
+   * @param paramsOfCurrentMethod the parameters of the current method
+   * @param invocation the method invocation node to check
+   * @param arg the argument of a method invocation node
+   */
+  private void analyzeCalledMethodsSetForArgAliasesToInferOwningParams(
+      Set<Obligation> obligations,
+      List<? extends VariableTree> paramsOfCurrentMethod,
+      MethodInvocationNode invocation,
+      Node arg) {
+
+    for (int i = 1; i < paramsOfCurrentMethod.size() + 1; i++) {
+      if (resourceLeakAtf.hasEmptyMustCallValue(paramsOfCurrentMethod.get(i - 1))) {
+        continue;
+      }
+      VariableTree currentMethodParamTree = paramsOfCurrentMethod.get(i - 1);
+      VariableElement currentMethodParamElt =
+          TreeUtils.elementFromDeclaration(currentMethodParamTree);
+      if (isParamAndArgAliased(obligations, arg, currentMethodParamElt)) {
+        JavaExpression paramJe = JavaExpression.fromVariableTree(currentMethodParamTree);
+        if (mustCallObligationSatisfied(invocation, currentMethodParamElt, paramJe)) {
+          addOwningToParam(i);
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns the set of resource aliases associated with the given argument node, by looking up the
+   * corresponding obligation in the given set of obligations.
+   *
+   * @param obligations the set of obligations to search in
+   * @param arg the argument node whose resource aliases are to be returned
+   * @return the resource aliases associated with the given argument node, or an empty set if the
+   *     node has none
+   */
+  private Set<ResourceAlias> getResourceAliasOfArgument(Set<Obligation> obligations, Node arg) {
```

Comment:
Why is the name "ofArgument"?  Argument of what?  Please make the name and documentation describe the semantics of this method on its own, rather than reflecting a fact about the current call sites of the method.